mitmproxy/websocket.py

```diff
@@ -0,0 +1,51 @@
+import time
+
+from typing import List
+
+from mitmproxy import flow
+from mitmproxy.types import serializable
+
+
+class WebSocketMessage(serializable.Serializable):
```

Comment:
Here we should retain whether the original data arrived as text or a binary frames. This changes the semantics of the content - the standard says that text frames when assembled always has to result in valid UTF-8, binary can obviously be anything. There will be differences in how the servers deal with this, and in how we should expose the content. Maybe even two classes - WebSocketBinaryMessage, WebSocketTextMessage.